From: Andreas Pehrson <apehrson@mozilla.com>
Date: Thu, 29 Aug 2024 08:31:00 +0000
Subject: Bug 1914837 - Always initialize DesktopCapturer::pid.
 r=webrtc-reviewers,ng

This member is patched on top of libwebrtc. New backends could be added from
under us. It needs to be initialized.

Differential Revision: https://phabricator.services.mozilla.com/D220265
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/fdf5b0a81b0c0e87d31c5963091681a0e5960678
---
 modules/desktop_capture/desktop_capturer.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/desktop_capture/desktop_capturer.h b/modules/desktop_capture/desktop_capturer.h
index 60c04723d6..fb3cc94f74 100644
--- a/modules/desktop_capture/desktop_capturer.h
+++ b/modules/desktop_capture/desktop_capturer.h
@@ -82,7 +82,7 @@ class RTC_EXPORT DesktopCapturer {
   struct Source {
     // The unique id to represent a Source of current DesktopCapturer.
     SourceId id;
-    pid_t pid;
+    pid_t pid = 0;
 
     // Title of the window or screen in UTF-8 encoding, maybe empty. This field
     // should not be used to identify a source.
